### PR TITLE
Refactor (iou_assigner): set low quality match as an option

### PR DESCRIPTION
**Pull request by ZwwWayne**
Add an option in iou assigners: `match_low_quality`.  
This option decides whether to allow quality matches. This is usually allowed for RPN and single stage detectors, but not allowed in the second stage.

### CODE REVIEW

File: mmdet/core/bbox/assigners/approx_max_iou_assigner.py [hunk 1]
```diff
@@ -27,6 +27,9 @@ class ApproxMaxIoUAssigner(MaxIoUAssigner):
             ignoring any bboxes.
         ignore_wrt_candidates (bool): Whether to compute the iof between
             `bboxes` and `gt_bboxes_ignore`, or the contrary.
+        match_low_quality (bool): Whether to allow quality matches. This is
```
Comment (hellock) [inline mmdet/core/bbox/assigners/approx_max_iou_assigner.py:30]: typo: allow low quality matches.

File: mmdet/core/bbox/assigners/approx_max_iou_assigner.py [hunk 2]
```diff
@@ -27,6 +27,9 @@ class ApproxMaxIoUAssigner(MaxIoUAssigner):
...
+            usually allowed for RPN and single stage detectors, but not allowed
+            in the second stage.
```
Comment (hellock) [inline mmdet/core/bbox/assigners/approx_max_iou_assigner.py:32]: May refer to step4 to have a brief description on what is `low quality match`.

File: mmdet/core/bbox/assigners/max_iou_assigner.py [hunk 3]
```diff
@@ -173,14 +179,20 @@ def assign_wrt_overlaps(self, overlaps, gt_labels=None):
...
-                    assigned_gt_inds[max_iou_inds] = i + 1
-                else:
-                    assigned_gt_inds[gt_argmax_overlaps[i]] = i + 1
+        if self.match_low_quality:
+            # low-quality matching will overwirte the assigned_gt_inds assigned
+            # in Step 3
+            # thus the assigned gt might not be the best one for the prediction
+            # e.g., bbox A has 0.9 and 0.8 iou with GT bbox 1 & 2
+            # GT bbox 1's & 2's gt_argmax_overlaps = A
+            # This might be the reason that it is not used in ROI Heads.
```
Comment (hellock) [inline mmdet/core/bbox/assigners/max_iou_assigner.py:188]: The example may be confusing.

File: mmdet/core/bbox/assigners/max_iou_assigner.py [hunk 4]
```diff
@@ -28,6 +28,9 @@ class MaxIoUAssigner(BaseAssigner):
             ignoring any bboxes.
         ignore_wrt_candidates (bool): Whether to compute the iof between
             `bboxes` and `gt_bboxes_ignore`, or the contrary.
+        match_low_quality (bool): Whether to allow low quality matches. This is
+            usually allowed for RPN and single stage detectors, but not allowed
+            in the second stage. Details are demonetrates in Step 4.
```
Comment (hellock) [inline mmdet/core/bbox/assigners/max_iou_assigner.py:33]: demonstrated

**Comment by xvjiarui** [comment]
Missing `cascade_rcnn`?

**Pull request merged**